### PR TITLE
Streams for tcp

### PR DESCRIPTION
**Pull request by t-constantin**
Removed textview text on receive from communication device
Merged TcpClient functions into TCPDevice file.
TCPDevice now expose socket's input/output stream

### CODE REVIEW

File: HiveAR/app/src/main/java/com/swarmus/hivear/activities/TestActivity.java [hunk 1]
```diff
@@ -150,51 +149,44 @@ private void switchCommunication(Fragment toShow) {
...
         }
 
         @Override
         public void onConnectError() {
-            currentCommunicationDevice.broadCastConnectionStatus(ConnectionStatus.notConnected);
+            tcpDevice.broadCastConnectionStatus(ConnectionStatus.notConnected);
         }
     };
 
     final UsbSerialInterface.UsbReadCallback usbReadCallback = (data) -> {
```
Comment (cquesnel) [inline HiveAR/app/src/main/java/com/swarmus/hivear/activities/TestActivity.java:188]: If the readCallback was removed in serialDevice, is this function still necessary?

File: HiveAR/app/src/main/java/com/swarmus/hivear/models/TCPDevice.java [hunk 2]
```diff
@@ -1,67 +1,142 @@
...
     public void setServerPort(int port) {this.serverPort=port;}
 
-    public class ConnectTask extends AsyncTask<String, String, TcpClient> {
-
-        @Override
-        protected TcpClient doInBackground(String... message) {
-
-            tcpClient = new TcpClient(listener, serverIP, serverPort);
-            tcpClient.run();
+    public InputStream getSocketInputStream()
```
Comment (cquesnel) [inline HiveAR/app/src/main/java/com/swarmus/hivear/models/TCPDevice.java:110]: Why have the same method twice? `getDataStream()` and `getSocketInputStream()`

File: HiveAR/app/src/main/java/com/swarmus/hivear/activities/TestActivity.java (hunk 1 above)
Comment (t-constantin) [reply HiveAR/app/src/main/java/com/swarmus/hivear/activities/TestActivity.java:188]: yes, it is used to send a signal to tell the ui that the connection is not active (icon become read on connect error)

File: HiveAR/app/src/main/java/com/swarmus/hivear/models/TCPDevice.java (hunk 2 above)
Comment (t-constantin) [reply HiveAR/app/src/main/java/com/swarmus/hivear/models/TCPDevice.java:110]: completely useless, removed.

**Review by cquesnel — APPROVED** [review]

**Pull request merged**